sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -340,6 +340,21 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
           // surrounded with single quotes, or there is a typo in the attribute name.
           case GetMapValue(map, key: Attribute) if isMapWithStringKey(map) && !key.resolved =>
             failUnresolvedAttribute(operator, key, "UNRESOLVED_MAP_KEY")
+
+          case e: Expression if e.containsPattern(LATERAL_COLUMN_ALIAS_REFERENCE) &&
+            operator.expressions.exists {
+              case a: Alias
+                  if e.collect { case l: LateralColumnAliasReference => l.nameParts.head }
+                    .contains(a.name) =>
+                a.exists(_.isInstanceOf[Generator])
+              case _ => false
+            } =>
```

Comment:
How about to put the condition check to a separate function, see `isMapWithStringKey` above for instance. This could improve readability, probably.

Reply:
fixed.